geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/planning/guard/GraduatedQueryGuard.scala

```diff
@@ -32,12 +35,25 @@ class GraduatedQueryGuard extends QueryInterceptor with LazyLogging {
       logger.info(s"This guard is disabled for schema '${sft.getTypeName}' via system property")
     }
     // let any errors bubble up and disable this guard
-    guardLimits = buildLimits(ConfigFactory.load().getConfigList(s"$ConfigPath.${sft.getTypeName}"))
+    guardLimits = buildLimits(ConfigFactory.load().getConfigList(s"$ConfigPath.${sft.getTypeName}"), sft)
     // this should be ensured during the loading of limits, but just double check so we can use `.last` safely below
     require(guardLimits.nonEmpty)
   }
 
-  override def rewrite(query: Query): Unit = {}
+  override def rewrite(query: Query): Unit = {
+    if (!disabled) {
+      val bounds = query.getFilter.accept(ExtractBoundsFilterVisitor.BOUNDS_VISITOR, null).asInstanceOf[Envelope]
```

Comment:
do you know if this operates on the default geom by default, or maybe we need to pass that in? technically there could be indices on other geometries, but i guess we don't know what index is being used yet at this point in the processing pipeline... maybe just using "any" geom would be best here (not sure if that's what it's currently doing or not)

Reply:
The geotools docs say 
> all the literal geometry instances involved if spatial operations - using geom.getEnvelopeInternal().
which I would interpret to mean it considers all geometries included in the query.